docs/contribute/logging.md

```diff
@@ -0,0 +1,22 @@
+# Logging
+
+`cloudcost-exporter` codebase has grown to the point where we need to have some form of structured logging. 
+The initial commit was introduced `db82ae9ccdfddc010492f4739724c8e67ef40851` with a fairly detailed messaged on the requirements and the approach.
+The very short is we needed:
+
+1. Structured logging with consistent labels across providers and collectors
+2. Ability to define a log level and runtime that limits the logs emitted
+
+With `slog` being part of Go's stdlib since 1.21, we decided to use it as the logging library with a wrapper so that we can get log levels as well.
+
+## Guidelines
+
+1. Every provider _must_ accept a `*slog.Logger` in the constructor
+1. Every collector _must_ accept a `*slog.Logger` in the constructor
+1. Each provider and collect _must_ add a `provider` or `collector` group when initializing using the [slog.WithGroup](https://pkg.go.dev/golang.org/x/exp/slog#Logger.WithGroup) method
+1. Always prefer to use the `logger.WithAttr(...)` method to add structured data to the log message for both performance and consistency
```

Comment:
Personally, I prefer things like `<loggingPackage>.Info(message, args...)`, it flows a lot more intuitively to me.

For example:

```go
		log.Info("Starting server",
			slog.String("address", cfg.Server.Address),
			slog.String("path", cfg.Server.Path))
```

just looks better than:

```go
		log.LogAttrs(ctx, slog.LevelInfo, "Starting server",
			slog.String("address", cfg.Server.Address),
			slog.String("path", cfg.Server.Path))
```

in my opinion.

Reply:
if `ctx` is important (doesn't seem to show up in the logs, but I suppose could be used for things like timeouts, we could use `InfoCtx`

Reply:
So unfortunately this isn't just a semantic difference on looks. There's both a performance reason for using `logAttrs` and gaining access to context, which is simply not possible with using the pre built `log.(Debug|Info|Warn|Error`. See https://go.dev/blog/slog and search for Performance for a better overview on why.

Reply:
Let me update that line to link out to the go blog post, as I feel this is going to be a somewhat controversial/highly opinionated topic 😅

Reply:
👍